eng/validation/build-arcadewithrepo.ps1

```diff
@@ -0,0 +1,265 @@
+    [string] $azdoToken)
+{
+    $uri = "https://dev.azure.com/${azdoOrg}/${azdoProject}/_apis/build/builds/${buildId}?api-version=5.1"
+    $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(":${azdoToken}"))
+    $headers = @{"Authorization"="Basic $base64AuthInfo"}
+    $content = Invoke-WebRequest -Uri $uri -Headers $headers -ContentType "application/json" -Method Get 
+    return ($content | ConvertFrom-Json).status
+}
+
+function Get-BuildResult(
+    [string] $azdoOrg, 
+    [string] $azdoProject,
+    [int] $buildId,
+    [string] $azdoToken)
+{
+    $uri = "https://dev.azure.com/${azdoOrg}/${azdoProject}/_apis/build/builds/${buildId}?api-version=5.1"
+    $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(":${azdoToken}"))
+    $headers = @{"Authorization"="Basic $base64AuthInfo"}
+    $content = Invoke-WebRequest -Uri $uri -Headers $headers -ContentType "application/json" -Method Get 
+    return ($content | ConvertFrom-Json).result
+}
+
+function Get-Github-RepoAuthUri($repoName)
+{
+    "https://${githubUser}:${githubPAT}@github.com/${githubOrg}/${repoName}"
+}
+
+function GitHub-Clone($repoName) 
+{
+    $authUri = Get-Github-RepoAuthUri $repoName
+    & git clone $authUri $(Get-Repo-Location $repoName)
+    Push-Location -Path $(Get-Repo-Location $repoName)
+    & git config user.email "${githubUser}@test.com"
+    & git config user.name $githubUser
+    Pop-Location
+}
+
+function Get-Repo-Location($repoName)
+{
+    "$testRoot\$repoName"
+}
+
+function Git-Command($repoName) {
+    Push-Location -Path $(Get-Repo-Location($repoName))
+    try {
+        $gitParams = $args
+        if ($gitParams.GetType().Name -ne "Object[]") {
+            $gitParams = $gitParams.ToString().Split(" ")
+        }
+        Write-Host "Running 'git $gitParams' from $(Get-Location)"
+        $commandOutput = & git @gitParams; if ($LASTEXITCODE -ne 0) { throw "Git exited with exit code: $LASTEXITCODE" } else { $commandOutput }
+        $commandOutput
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+## If able to retrieve a build, get the SHA that it was built from
+$sha = Get-LastKnownGoodBuildSha -azdoOrg $azdoOrg -azdoProject $azdoProject -buildDefinitionId $buildDefinitionId -azdoToken $azdoToken
+
+## Clone the repo from git
+GitHub-Clone $targetRepoName
+
+## Create a branch from the repo with the given SHA. 
+$targetBranch = "dev/" + $githubUser + "/arcade-" + $arcadeSdkVersion
+$githubUri = Get-Github-RepoAuthUri $targetRepoName
+
+$darcBranchName = "refs/heads/" + $targetBranch
+$darcRepoName = "https://github.com/${githubOrg}/${targetRepoName}"
+
+# check to see if branch exists and clean it up if it does
+$branchExists = Git-Command $targetRepoName ls-remote --heads $githubUri refs/heads/$targetBranch
+if($null -ne $branchExists)
+{
+    & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcRepoName --github-pat $githubPAT --password $barToken
```

Comment:
We probably want to make sure that this doesn't fail the script. There's a (very very short) race where someone could delete the branch right before we check.

Same with default channel, someone could've deleted it.

If we fail to delete them we should inform, but not fail.

Reply:
Are there specific errors to look for from darc, or would wrapping this in a try/catch be sufficient?

Reply:
Try/catch is basically what I was thinking

Reply:
Did that here and also on the clean up at the end of the script just in case.